### PR TITLE
Connection override fix

### PR DESCRIPTION
**Pull request by singankit**
# Description

Connection override does not work when running a flow after first run.

```python

flow = load_flow(...)

flow.context.connections = {
            "query_llm": {
                "connection": AzureOpenAIConnection(
                    api_base=model_config.api_base,
                    api_key=model_config.api_key,
                    api_version=model_config.api_version,
                    api_type="azure"
                ),
                "deployment_name": deployment_name,
            }

flow(...) # This work
flow(....) # This fails

```

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow-devkit/promptflow/_sdk/entities/_flows/_flow_context_resolver.py [hunk 1]
```diff
@@ -82,7 +83,7 @@ def _resolve_connections(self, flow_context: FlowContext) -> "FlowContextResolve
 
         overwrite_connections(
             flow_dag=self.flow_dag,
-            connections=flow_context.connections,
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/entities/_flows/_flow_context_resolver.py:85]: add some test for this?

File: src/promptflow-devkit/promptflow/_sdk/entities/_flows/_flow_context_resolver.py [hunk 2]
```diff
@@ -82,7 +83,7 @@ def _resolve_connections(self, flow_context: FlowContext) -> "FlowContextResolve
...
+            connections=copy.deepcopy(flow_context.connections),
             working_dir=self.working_dir,
         )
         return self
```
Comment (D-W-) [inline src/promptflow-devkit/promptflow/_sdk/entities/_flows/_flow_context_resolver.py:89]: Maybe add a test to guard this?

File: src/promptflow-devkit/promptflow/_sdk/entities/_flows/_flow_context_resolver.py (hunk 1 above)
Comment (singankit) [reply src/promptflow-devkit/promptflow/_sdk/entities/_flows/_flow_context_resolver.py:85]: Added test for it

File: src/promptflow-devkit/promptflow/_sdk/entities/_flows/_flow_context_resolver.py (hunk 2 above)
Comment (singankit) [reply src/promptflow-devkit/promptflow/_sdk/entities/_flows/_flow_context_resolver.py:89]: Yes added test for it

**Review by wangchao1230 — DISMISSED** [review]

**Review by D-W- — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by D-W- — APPROVED** [review]

**Pull request merged**